src/main/java/bussiness_layer/services/UserService.java

```diff
@@ -19,7 +19,7 @@
 
     @Override
     public UserDTO getUserByUsername(String username) {
-        User user = userRepo.findOneByUsername(username);
+        User user = userRepo.findByUsername(username);
```

Comment:
I still consider that it is a better approach to return Optional<Username> from repository and here to check if that optional is present and then get the related object. Java programmers be like :)